NXP-32667: allow configuring the pod and container securityContext

nichiporenko opened the pull request:
Issue: #[HXAI-3241](https://hyland.atlassian.net/browse/HXAI-3241)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/nuxeo/nuxeo-helm-chart?pullRequest=59) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/nuxeo/nuxeo-helm-chart?pullRequest=59) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/nuxeo/nuxeo-helm-chart?pullRequest=59) it.</sub>

nichiporenko commented:
> Hi, thanks for this PR that sounds legitimate.
> 
> As the Jira project used for this repository is [NXP](https://jira.nuxeo.com/projects/NXP) in Nuxeo Jira, can you please rephrase you commit message to use it. I've created [NXP-32667](https://jira.nuxeo.com/browse/NXP-32667) for the occasion, similarly to the [NXP-32286](https://jira.nuxeo.com/browse/NXP-32286) Jira task used for the latest [commit](https://github.com/nuxeo/nuxeo-helm-chart/commit/ec3e4c1876e04c9260650ecc47793d9504ca0d9d).
> 
> Regarding the changes, I would suggest to also handle securityContext at the pod level, currently hardcoded to:
> 
> ```
> securityContext:
>   fsGroup: 1000
> ```
> 
> See:
> 
> https://github.com/nuxeo/nuxeo-helm-chart/blob/ec3e4c1876e04c9260650ecc47793d9504ca0d9d/nuxeo/templates/deployment.yaml#L152
> 
> 
> So, in values.yaml, we could have something like:
> ```
> podSecurityContext:
>   fsGroup: 1000
> containerSecurityContext:
>   capabilities:
>     drop:
>       - ALL
> ```
> 
> Thanks

Hi, thank you. Updated

nichiporenko commented:
Hi @ataillefer and other colleagues from Nuxeo. I've updated the PR. Could you please re-review it again?
Removed dropping all capabilities from the containerSecurityContext by default. We will pass the value there through the [values.yaml file in hxai-insight-nuxeo](https://github.com/HylandSoftware/hxai-insight-nuxeo/blob/master/charts/values.yaml)
